src/libs/actions/BankAccounts.ts

```diff
@@ -174,103 +192,86 @@ function addPersonalBankAccount(account) {
     API.write(commandName, parameters, onyxData);
 }
 
-function deletePaymentBankAccount(bankAccountID) {
-    API.write(
-        'DeletePaymentBankAccount',
-        {
-            bankAccountID,
-        },
-        {
-            optimisticData: [
-                {
-                    onyxMethod: Onyx.METHOD.MERGE,
-                    key: `${ONYXKEYS.BANK_ACCOUNT_LIST}`,
-                    value: {[bankAccountID]: {pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE}},
-                },
-            ],
-
-            // Sometimes pusher updates aren't received when we close the App while still offline,
-            // so we are setting the bankAccount to null here to ensure that it gets cleared out once we come back online.
-            successData: [
-                {
-                    onyxMethod: Onyx.METHOD.MERGE,
-                    key: `${ONYXKEYS.BANK_ACCOUNT_LIST}`,
-                    value: {[bankAccountID]: null},
-                },
-            ],
-        },
-    );
+function deletePaymentBankAccount(bankAccountID: number) {
+    type DeletePaymentBankAccountParams = {bankAccountID: number};
+
+    const parameters: DeletePaymentBankAccountParams = {bankAccountID};
+
+    API.write('DeletePaymentBankAccount', parameters, {
+        optimisticData: [
```

Comment:
API migration is not merged yet, please extract `optimisticData`, `successData` and `failureData` to this format:
```ts
    const optimisticData: OnyxUpdate[] = [...];
    const successData: OnyxUpdate[] = [...];
    const failureData: OnyxUpdate[] = [...];

    API.write('', parameters, {optimisticData, successData, failureData});
```

This will make it more type safe (same for other functions). Thanks!

Reply:
@blazejkustra Updated